cmake_modules/IcebergThirdpartyToolchain.cmake

```diff
@@ -86,6 +86,9 @@ function(resolve_arrow_dependency)
   set(ARROW_DEPENDENCY_SOURCE
       "BUNDLED"
       CACHE STRING "" FORCE)
+  set(ARROW_WITH_ZLIB
```

Comment:
We should also add `set(ZLIB_SOURCE "SYSTEM" CACHE STRING "" FORCE)`

Reply:
DONE